mitodl/micromasters
# Added learner-learner search page (#2690)

#### What are the relevant tickets?
Fixes #2512 

#### What's this PR do?
Adds a learner to learner search page.

#### How should this be manually tested?

You'll need to run `./manage.py recreate_index`.

- Staff search should work as it always has.
- Learners should be able to search only on name and filter only on location, course, and semester.

#### Where should the reviewer start?
(Optional)

#### Any background context you want to provide?
One of my main concerns around this work was ensuring we don't expose any sensitive profile data. The was accomplished by doing the following:

- Added a new doctype that only indexes and stores non-sensitive profile information. Learner-learner search filters to this doctype. This prevents the possibility of inferring private information through filtering or aggregates and generally ensures that sensitive data doesn't end up in `_source`.
- Added enforcement of filtering to non-privat eprofiles for the learner-learner search query.
- Added enforcement of `_source` filtering on the learner-learner search query in case anything ever slipped through in the future.

#### Screenshots (if appropriate)
![selection_046](https://cloud.githubusercontent.com/assets/28598/24269361/51660e82-0fe7-11e7-8424-c5434881b44c.png)


#### What GIF best describes this PR or how it makes you feel?
![](http://media1.giphy.com/media/j3F9SdJo4lIsM/200w.gif)
